projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -12,12 +12,89 @@ export function cloneArray(array, deep?: boolean) {
     }
     return arr;
 }
+
 /**
+ * Deep clones all first level keys of Obj2 and merges them to Obj1
+ * @param obj1 Object to merge into
+ * @param obj2 Object to merge from
+ * @returns Obj1 with merged cloned keys from Obj2
+ * @hidden
+ */
+export function mergeObjects(obj1: {}, obj2: {}) {
+    if (obj1 === null || obj1 === undefined) {
+        return this.cloneObject(obj2);
+    }
+
+    if (obj2 === null || obj2 === undefined) {
+        return obj1;
+    }
+
+    if (!isObject(obj1)) {
+        throw new Error(`Should provide objects to mergeObjects method. ${obj1} is not an object!`);
+    }
+
+    if (!isObject(obj2)) {
+        throw new Error(`Should provide objects to mergeObjects method. ${obj2} is not an object!`);
+    }
+
+    for (const key of Object.keys(obj2)) {
+        obj1[key] = this.cloneObject(obj2[key]);
+    }
+
+    return obj1;
+}
+
+/**
+ * Creates deep clone of provided value.
+ * Supports primitive values, dates and objects.
+ * If passed value is array returns shallow copy of the array.
+ * @param value value to clone
+ * @returns Deep copy of provided value
  *@hidden
  */
-export function cloneObject(object: any) {
-    return JSON.parse(JSON.stringify(object));
+export function cloneObject(value: any): any {
+    if (this.isDate(value)) {
+        return new Date(value.getTime());
+    }
+    if (Array.isArray(value)) {
+        return [...value];
+    }
+
+    if (value instanceof Map || value instanceof Set) {
+        return value;
+    }
+
+    if (this.isObject(value)) {
+        const result = {};
+
+        for (const key of Object.keys(value)) {
+            result[key] = this.cloneObject(value[key]);
+        }
+        return result;
+    }
+    return value;
+}
+
+/**
+ * Checks if provided variable is Object
+ * @param value Value to check
+ * @returns true if provided variable is Object
+ *@hidden
+ */
+export function isObject(value: any): boolean {
+    return value !== null && value !== undefined && value.toString() === '[object Object]';
+}
+
+/**
+ * Checks if provided variable is Date
+ * @param value Value to check
+ * @returns true if provided variable is Date
+ *@hidden
+ */
+export function isDate(value: any) {
+    return value && Object.prototype.toString.call(value) === '[object Date]' && !isNaN(value);
```

Comment:
```suggestion
    return Object.prototype.toString.call(value) === '[object Date]';
```
Should be enough here. This is a copy only, the validity of the date is not relevant.

Reply:
Umm no it won't.. unless it's different in IE